coordinates/source/dhtslib/package.d

```diff
@@ -342,75 +357,60 @@ struct Interval(CoordSystem cs)
         }
     }
 
-    auto getMinStart(Interval!cs other)
+    /// get minimum start value between this interval
+    /// and other interval
+    auto getMinStart(Interval!cs other) @nogc
     {
         return this.start < other.start ? this.start : other.start;
     }
 
-    auto getMaxStart(Interval!cs other)
+    /// get maximum start value between this interval
+    /// and other interval
+    auto getMaxStart(Interval!cs other) @nogc
     {
         return this.start > other.start ? this.start : other.start;
     }
 
-    auto getMinEnd(Interval!cs other)
+    /// get minimum end value between this interval
+    /// and other interval
+    auto getMinEnd(Interval!cs other) @nogc
     {
         return this.end < other.end ? this.end : other.end;
     }
     
-    auto getMaxEnd(Interval!cs other)
+    /// get maximum end value between this interval
+    /// and other interval
+    auto getMaxEnd(Interval!cs other) @nogc
     {
         return this.end > other.end ? this.end : other.end;
     }
 
     /// set operators for intersect, union, and difference
-    auto opBinary(string op)(Interval!cs other)
+    auto opBinary(string op)(Interval!cs other) @nogc
```

Comment:
we don't actually have a difference operator. you could use e.g. ^ but honestly probably not that useful of an operation.

Change documentation per Arne's suggestion:

/// Set operators for interval intersection  (`|`) and union (`&`).
/// If the intervals are non-overlapping, intersection returns an empty interval (*fill in how you've represented empty interval*)
/// while union returns the convex hull.

Reply:
I did update the union function implementation, but I can update it here as well.

Reply:
oh ok didn't see that, sorry

Reply:
just updated with latest commit